app/helpers/wowza_helper.rb

```diff
@@ -0,0 +1,44 @@
+module WowzaHelper
+  #---
+  # Builds Wowza URL
+  #
+  # @param obj_id
+  # @param file_set_id
+  # @param fedora_file
+  # @media_type: 'audio', 'video'
+  # @return string or nil
+  #---
+
+  def grabWowzaURL(obj_id, file_set_id, fedora_file, media_type, request_ip)
+    encrypted = encrypt_stream_name( obj_id, file_set_id, fedora_file, media_type, request_ip )
+    return Rails.configuration.wowza_baseurl + encrypted
+  end
+
+
+  ## video stream name encryption
+  def encrypt_stream_name( obj_id, file_set_id, fedora_file, media_type, ip )
+
+    # random nonce
+    nonce=rand(36**16).to_s(36)
+    while nonce.length < 16 do
+      nonce += "x"
+    end
+
+    # load key from environment variable
+    key = ENV.fetch('APPS_DHH_STREAMING_KEY') {'xxxxxxxxxxxxxxxx'}
+    
+    # encrypt
+    str="#{obj_id} #{file_set_id} #{fedora_file} #{media_type} #{ip}"
```

Comment:
This is just an idea to consider. But as far as I can tell, it seems that all of the arguments passed to this method are being used only once (here) to create the string to pass to the cipher to encrypt. Does it make sense to consolidate them into a single string to pass into this method as a single argument?